ddtrace/context.py

```diff
@@ -241,3 +240,39 @@ def get(self):
             self._locals.context = ctx
 
         return ctx
+
+
+class _ContextVarContext(object):
+    """
+    _ContextVarContext can be used in place of the ThreadLocalContext for Python
+    3.7 and above to manage different ``Context`` objects for each thread and
+    async task.
+    """
+
+    def __init__(self):
+        self.token = None
+
+    def _has_active_context(self):
+        ctx = DATADOG_CONTEXT.get()
+        return ctx is not None
+
+    def set(self, ctx):
+        self.token = DATADOG_CONTEXT.set(ctx)
+
+    def get(self):
+        ctx = DATADOG_CONTEXT.get()
+        if not ctx:
+            ctx = Context()
+            self.set(ctx)
+
+        return ctx
+
+
+try:
+    from contextvars import ContextVar
+
+    DATADOG_CONTEXT = ContextVar('datadog_contextvar', default=None)
```

Comment:
I'm not seeing why you need this global variable. Why don't you just create a `ContextVar` inside the `_ContextVarContext.__init__` method and use it?

Reply:
This is the usage following documentation:

> Important: Context Variables should be created at the top module level and never in closures. Context objects hold strong references to context variables which prevents context variables from being properly garbage collected.

https://docs.python.org/3/library/contextvars.html#contextvars.ContextVar

Reply:
Maybe an alternative could be.

```python
class ContextVarContext(object):
    _CONTEXT_VARS = ContextVar('datadog_contextvar', default=

    @classmethod 
    def reset_context(cls):
        cls._CONTEXT_VARS.set(None)

    # Do everything else as before
```